spec.html

```diff
@@ -34198,11 +34198,29 @@ <h1>
             1. Let _intMaxLength_ be ℝ(? ToLength(_maxLength_)).
             1. Let _stringLength_ be the length of _S_.
             1. If _intMaxLength_ ≤ _stringLength_, return _S_.
-            1. If _fillString_ is *undefined*, let _filler_ be the String value consisting solely of the code unit 0x0020 (SPACE).
-            1. Else, let _filler_ be ? ToString(_fillString_).
-            1. If _filler_ is the empty String, return _S_.
-            1. Let _fillLen_ be _intMaxLength_ - _stringLength_.
-            1. Let _truncatedStringFiller_ be the String value consisting of repeated concatenations of _filler_ truncated to length _fillLen_.
+            1. If _fillString_ is not *undefined*, set _fillString_ to ? ToString(_fillString_).
+            1. Return StringPad(_S_, _intMaxLength_, _fillString_, _placement_).
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-stringpad" type="abstract operation">
+          <h1>
+            StringPad (
+              _S_: a String,
+              _maxLength_: a non-negative integer,
+              _fillString_: a String or *undefined*,
```

Comment:
This should just be `a String`; the `undefined` case should be hoisted into `StringPaddingBuiltinsImpl`. Otherwise LGTM; I'll push up a commit with that change so it doesn't need to block.